### PR TITLE
storage/mysql: Replace enable_mysql_source with max_mysql_connections var

### PR DESCRIPTION
**Pull request by rjobanp**
### Motivation

Parity with other source types, and the ability to query this variable from the Console (via SQL). See https://materializeinc.slack.com/archives/C01CFKM1QRF/p1715228421858309?thread_ts=1715209780.830859&cid=C01CFKM1QRF for more context

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by def- — COMMENTED** [review]
I triggered nightly just in case this has some weird side effects with upgrades: https://buildkite.com/materialize/nightly/builds/8207

**Review by def- — CHANGES_REQUESTED** [review]
There is a bunch of failures in nightly, but they should all be fixable by bumping the limit. Ping me if you want me to fix those errors as part of this PR.

File: src/sql/src/session/vars/definitions.rs [hunk 1]
```diff
@@ -432,6 +432,13 @@ pub static MAX_POSTGRES_CONNECTIONS: VarDefinition = VarDefinition::new(
     false,
 );
 
+pub static MAX_MYSQL_CONNECTIONS: VarDefinition = VarDefinition::new(
+    "max_mysql_connections",
+    value!(u32; 0),
```
Comment (benesch) [inline src/sql/src/session/vars/definitions.rs:437]: Your life will probably be much easier if you default this to 1000, and then use LD in production to set it back to 0 by default for production environments.

Comment (rjobanp) [reply src/sql/src/session/vars/definitions.rs:437]: ah great idea - will do

Comment (rjobanp) [reply src/sql/src/session/vars/definitions.rs:437]: sounds good, set up the LD flag in prod to do so

**Review by def- — APPROVED** [review]

**Pull request merged**